util/src/main/java/tc/oc/pgm/util/text/types/PlayerComponent.java

```diff
@@ -38,32 +38,49 @@ static Component of(@Nullable Player player, String defName, NameStyle style) {
       return component.build();
     }
 
-    Component usernameComponent;
+    // For name styles that don't allow vanished, make vanished appear offline
+    if (!style.showVanish && isVanished(player)) {
```

Comment:
I think we need to clean up this code. Maybe separate methods for each `NameStyle`?

Reply:
I agree, with the previous refactor I tried to keep it similar to the old `PersonalizedPlayer` class, but imo would be better to separate into different methods as you suggested 👍 

I’ll take care of this in a bit.